File: dynamic_listings/C++/default/computer_science/bubblesort.cpp
void bubbleSort(int[] A){
    n = A.size();
    // Traverse Through all Elements
    for (i = 0; i < n; ++i) {
        // Last i elements are in place due to the nature of the sort
        for (j = 0; j < n - i - 1; ++j) {
            // Swap if the element found is greater than the next element
            if (A[j] > A[j+1]){
                int tmp = A[j];
                A[j] = A[j+1];
                A[j+1] = tmp;
            }
        }
    }
}

File: dynamic_listings/rust/default/computer_science/bubblesort.rs
# #![allow(dead_code)]
fn bubble_sort(a: &mut [u32]) {
    let n = a.len();
    // Traverse Through all Elements
    for x in 0..n {
        // Last x elements are in place due to the nature of the sort
        for y in 0..(n - x - 1) {
            // Swap if the element found is greater than the next element
            if a[y] > a[y + 1] {
                let tmp = a[y];
                a[y] = a[y + 1];
                a[y + 1] = tmp;
            }
        }
    }
}
